orca-mine/src/main/groovy/com/netflix/spinnaker/orca/mine/pipeline/MonitorCanaryStage.groovy

```diff
@@ -49,8 +49,18 @@ class MonitorCanaryStage implements StageDefinitionBuilder, CancellableStage {
 
   @Override
   CancellableStage.Result cancel(Stage stage) {
-    def cancelCanaryResults = mineService.cancelCanary(stage.context.canary.id as String, "Pipeline execution (${stage.execution?.id}) canceled")
-    log.info("Canceled canary in mine (canaryId: ${stage.context.canary.id}, stageId: ${stage.id}, executionId: ${stage.execution.id})")
+    def cancelCanaryResults = [:]
+    def canaryId = stage.context.canary?.id as String
+
+    try {
+      if (canaryId) {
+        // will not have a `canaryId` if the failure occured prior to registration
+        cancelCanaryResults = mineService.cancelCanary(canaryId, "Pipeline execution (${stage.execution?.id}) canceled")
+        log.info("Canceled canary in mine (canaryId: ${canaryId}, stageId: ${stage.id}, executionId: ${stage.execution.id})")
+      }
+    } catch (Exception e) {
+      log.error("Unable to cancel canary '${canaryId}' in mine", e)
```

Comment:
The real point here is to ensure that launched canary clusters are cleaned up upon failure ... it would seem preferable that a failure to cancel in `mine` did NOT prevent cleanup.